api/services/account_service.py

```diff
@@ -277,18 +257,24 @@ def get_current_tenant_by_account(account: Account):
     @staticmethod
     def switch_tenant(account: Account, tenant_id: int = None) -> None:
         """Switch the current workspace for the account"""
-        if not tenant_id:
-            tenant_account_join = TenantAccountJoin.query.filter_by(account_id=account.id).first()
-        else:
-            tenant_account_join = TenantAccountJoin.query.filter_by(account_id=account.id, tenant_id=tenant_id).first()
 
-        # Check if the tenant exists and the account is a member of the tenant
-        if not tenant_account_join:
-            raise AccountNotLinkTenantError("Tenant not found or account is not a member of the tenant.")
+        with db.session.begin():
+            try:
+                tenant_account_join = TenantAccountJoin.query.filter_by(account_id=account.id, tenant_id=tenant_id).first()
+                TenantAccountJoin.query.filter_by(account_id=account.id).update({'current': False})
 
-        # Set the current tenant for the account
-        account.current_tenant_id = tenant_account_join.tenant_id
-        session['workspace_id'] = account.current_tenant.id
+                # Check if the tenant exists and the account is a member of the tenant
+                if not tenant_account_join:
+                    raise AccountNotLinkTenantError("Tenant not found or account is not a member of the tenant.")
```

Comment:
Better to move to in front of starting the db transaction.
If no tenant account was found, why not just return the error and pervert create and rollback the transaction?